flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/BufferWritingResultPartition.java

```diff
@@ -65,7 +65,7 @@ public abstract class BufferWritingResultPartition extends ResultPartition {
 
     private TimerGauge hardBackPressuredTimeMsPerSecond = new TimerGauge();
 
-    private long totalWrittenBytes;
+    protected long totalWrittenBytes;
```

Comment:
Ideally, `totalWrittenBytes` should be changed only in one class `BufferWritingResultPartition`. So maybe we should forbid `inherited classes` to write to `subpartitions` directly. But honestly, I don't know how to do it easily since the current implementation of `inherited classes` use `subpartitions` a lot and we can not change `subpartitions` visibility from `protected` to `private` . I mean we can create `addToSubpartition` method in `BufferWritingResultPartition` and use it everywhere(and maybe it will be better than now) but it still doesn't protect us from possible future bugs if somebody decides to use `subpartitions#add` directly in `inherited classes`.

Anyway, if you don't think that it will make things worse, maybe we indeed for now create the `add`(or `addToSubpartition`) method in `BufferWritingResultPartition` and will use it everywhere. At least it will reduce the number of places where we update `totalWrittenBytes`. Unfortunatelly, I didn't come up with any other solution until now.

Reply:
Hi @akalash , thanks a lot for your very patient review again.

>  I mean we can create addToSubpartition method in BufferWritingResultPartition and use it everywhere(and maybe it will be better than now)

I think it will be better than now as well, and this is the only better solution I can think of right now.

Updated by the last commit.